### ISSUE TITLE
Assigning criticals to non-empty slots freezes the entire client

### ISSUE THREAD

**Issue by Flibjib** [MegaMek/mekhq#536]
When modifying a unit in the MekLab, assigning a critical to slot that's already filled causes an error pop-up to appear, but the entire client is unresponsive.

I have to force quit the application to get out.

**Comment by HammerGS** [comment]
We need the log file and cpnx file to troubleshoot.

**Comment by Flibjib** [comment]
[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/1485957/mekhqlog.txt)

[Wolven Winter29921213.cpnx.zip](https://github.com/MegaMek/mekhq/files/1485958/Wolven.Winter29921213.cpnx.zip)

Attempted operation was simply to add an additional medium laser over top of on the previously present ones on a stock Grasshopper

**Comment by HammerGS** [comment]
NPE from Log.

```
megamek.common.LocationFullException: Medium Laser does not fit there in RA on Grasshopper GHR-5H Mk II (self)
	at megameklab.com.util.Mech.CriticalTransferHandler.addEquipmentMech(CriticalTransferHandler.java:320)
	at megameklab.com.util.Mech.CriticalTransferHandler.importData(CriticalTransferHandler.java:456)
	at javax.swing.TransferHandler$DropHandler.drop(TransferHandler.java:1544)
	at java.awt.dnd.DropTarget.drop(DropTarget.java:455)
	at javax.swing.TransferHandler$SwingDropTarget.drop(TransferHandler.java:1282)
	at sun.awt.dnd.SunDropTargetContextPeer.processDropMessage(SunDropTargetContextPeer.java:538)
	at sun.lwawt.macosx.CDropTargetContextPeer.processDropMessage(CDropTargetContextPeer.java:143)
	at sun.awt.dnd.SunDropTargetContextPeer$EventDispatcher.dispatchDropEvent(SunDropTargetContextPeer.java:852)
	at sun.awt.dnd.SunDropTargetContextPeer$EventDispatcher.dispatchEvent(SunDropTargetContextPeer.java:776)
	at sun.awt.dnd.SunDropTargetEvent.dispatch(SunDropTargetEvent.java:48)
	at java.awt.Component.dispatchEventImpl(Component.java:4746)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
```

**Comment by neoancient** [comment]
The campaign file has a custom unit that prevented it from being loaded.

Tried using a different campaign file and could not reproduce in the current version or in 0.43.4, which is the version that created the campaign file provided.

The stack trace in the log file was produced from an exception that was caught and handled. I removed that stack trace output a couple weeks ago because the popup that informs the user is sufficient.
